lib/AST/ASTPrinter.cpp

```diff
@@ -2379,9 +2379,43 @@ void PrintAST::visitStructDecl(StructDecl *decl) {
   }
 }
 
+/// Determines if a class has any designated initializers that are inaccessible
+/// to classes outside this module. If that's the case, then the subclass cannot
+/// declare its own designated initializers and will not inherit the convenience
+/// initializers from the superclass.
+static bool hasInaccessibleDesignatedInit(ClassDecl *decl) {
+  // We only care for classes that show up in module interfaces.
+  if (!isPublicOrUsableFromInline(decl))
+    return false;
```

Comment:
I don't think this is necessary to check; if we're printing the class, we already know we want this.